sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.models.ts

```diff
@@ -0,0 +1,248 @@
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  studentSINMatch: boolean;
+  /**
+   * Has correct institution code.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsIn([true], {
+    message: "The institution code provided is incorrect.",
+  })
+  hasCorrectInstitutionCode: boolean;
+  /**
+   * Application Status.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsEnum(ApplicationStatus)
+  @IsIn([ApplicationStatus.Completed], {
+    message: `The application is not in the ${ApplicationStatus.Completed} status.`,
+  })
+  applicationStatus: ApplicationStatus;
+  /**
+   * Application archived.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsIn([false], {
+    message: `Application is already archived and cannot be withdrawn.`,
+  })
+  isArchived: boolean;
+  /**
+   * Has previously been withdrawn.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsIn([false], {
+    message: (validationArguments: ValidationArguments) =>
+      `The application is already withdrawn with the date: ${validationArguments.object["withdrawalDate"]}.`,
+    context: ValidationContext.CreateWarning(
+      ApplicationWithdrawalValidationWarnings.HasPreviouslyBeenWithdrawn,
+    ),
+  })
+  hasPreviouslyBeenWithdrawn: boolean;
+  /**
+   * Is record match .ie. SIN, application number and institution must match a record in database.
+   */
+  @ValidateIf(
+    (object: ApplicationBulkWithdrawalImportBusinessValidationModel) =>
+      object.applicationFound,
+  )
+  @IsIn([true], {
+    message: "The record was not found and will be skipped.",
+    context: ValidationContext.CreateWarning(
+      ApplicationWithdrawalValidationWarnings.NoMatchingRecordFound,
+    ),
+  })
+  isRecordMatch: boolean;
+}
+
+/**
+ * Application bulk withdrawal header and data models.
+ */
+export interface FileData {
+  header: ApplicationBulkWithdrawalHeader;
+  applicationWithdrawalModels: ApplicationWithdrawalImportTextModel[];
+}
+
+/**
+ * Validation warning with a unique type and
+ * a user-friendly message to be displayed.
+ */
+export interface ValidationResult {
+  typeCode: ApplicationWithdrawalValidationWarnings;
+  message: string;
+}
+
+/**
+ * Results of the application bulk withdrawal model validation.
+ */
+export interface ApplicationWithdrawalValidationResult {
+  /**
+   * Record index in the list of records. Headers are not considered.
+   */
+  index: number;
+  /**
+   * Validated application bulk withdrawal model.
+   */
+  applicationBulkWithdrawalBusinessValidationModel: ApplicationBulkWithdrawalImportBusinessValidationModel;
```

Comment:
Can we have `applicationBulkWithdrawalBusinessValidationModel` renamed to `validationModel`?